diff --git a/process.c b/process.c
index 674f05dd6c..f52d6cb6a5 100644
--- a/process.c
+++ b/process.c
@@ -116,6 +116,7 @@ int initgroups(const char *, rb_gid_t);
 #include "ruby/util.h"
 #include "vm_core.h"
 #include "ruby/ractor.h"
+#include <emscripten.h>
 
 /* define system APIs */
 #ifdef _WIN32
@@ -8605,6 +8606,21 @@ proc_rb_f_kill(int c, const VALUE *v, VALUE _)
     return rb_f_kill(c, v);
 }
 
+static VALUE
+rb_emscripten_run_script(VALUE self, VALUE str)
+{
+    const char *js = StringValueCStr(str);
+    emscripten_run_script(js);
+    return Qnil;
+}
+
+static VALUE
+rb_emscripten_run_script_int(VALUE self, VALUE str)
+{
+    const char *js = StringValueCStr(str);
+    return INT2NUM(emscripten_run_script_int(js));
+}
+
 VALUE rb_mProcess;
 static VALUE rb_mProcUID;
 static VALUE rb_mProcGID;
@@ -8633,6 +8649,8 @@ InitVM_process(void)
     rb_define_global_function("sleep", rb_f_sleep, -1);
     rb_define_global_function("exit", f_exit, -1);
     rb_define_global_function("abort", f_abort, -1);
+    rb_define_global_function("emscripten_run_script", rb_emscripten_run_script, 1);
+    rb_define_global_function("emscripten_run_script_int", rb_emscripten_run_script_int, 1);
 
     rb_mProcess = rb_define_module("Process");
 
